packages/server/src/fhir/operations/export.ts

```diff
@@ -44,21 +44,28 @@ export async function bulkExportHandler(req: Request, res: Response): Promise<vo
   res.set('Content-Location', `${baseUrl}fhir/R4/bulkdata/export/${bulkDataExport.id}`).status(202).json(accepted);
 }
 
-async function exportResourceType(exporter: BulkExporter, project: Project, resourceType: ResourceType): Promise<void> {
+export async function exportResourceType(
+  exporter: BulkExporter,
+  resourceType: ResourceType,
+  maxResources = 1000
+): Promise<void> {
   const repo = exporter.repo;
   const hasMore = true;
+  let offset = 0;
   while (hasMore) {
     const bundle = await repo.search({
       resourceType,
-      count: 1000,
+      count: maxResources,
+      offset,
     });
     if (!bundle.entry || bundle.entry.length === 0) {
       break;
     }
 
     const writes = [];
     for (const entry of bundle.entry) {
-      if (entry.resource?.id) {
+      if (entry.resource) {
+        await exporter.getWriter(entry.resource.resourceType);
```

Comment:
this initializes the writer for each resource type

Reply:
This is still kinda clunky.

Can we align with the the `groupexport.ts` implementation, and call `await exporter.writeBundle`?

https://github.com/medplum/medplum/blob/main/packages/server/src/fhir/operations/groupexport.ts#LL47C11-L47C46

Then remove the `writes` and the `Promise.all`.